Make sure we don't relocate the weight name buffer

yinghai opened the pull request:
Summary: `onnxTensorDescriptorV1.name` points to the string buffer. We use a vector of strings to serve as the storage. This means we cannot reallocate the vector because that may invalidate the `onnxTensorDescriptorV1.name` pointers. Solution is to reserve a large enough vector so that it won't reallocate.

Differential Revision: D9381838

bddppq reviewed (approved):
Mind adding a comment?

Commit:
Make sure we don't relocate the weight name buffer (#10630)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/10630

`onnxTensorDescriptorV1.name` points to the string buffer. We use a vector of strings to serve as the storage. This means we cannot reallocate the vector because that may invalidate the `onnxTensorDescriptorV1.name` pointers. Solution is to reserve a large enough vector so that it won't reallocate.

Differential Revision: D9381838

fbshipit-source-id: 059f4e947e23d5ef36577ab6ab6a32e3e0042b62
